src/js/jsx/search/SearchBar.jsx

```diff
@@ -88,50 +91,188 @@ define(function (require, exports, module) {
+            } else {
+                iconID += layer.kind;
+            }
+            
+            return iconID;
+        },
 
         /**
-         * Make list of items and headers to be used as dropdown options
+         * Get the layer ancestry
+         *
+         * @private
+         * @param {Layer} layer
+         * @return {string}
+         */
+        _formatLayerAncestry: function (layer) {
+            var layerTree = this.getFlux().store("application").getCurrentDocument().layers,
+                ancestors = layerTree.ancestors(layer),
+                ancestorNames = ancestors.first().name;
+
+            ancestors.skip(1).forEach(function (ancestor) {
+                ancestorNames += ("/" + ancestor.name);
+            });
+            return ancestorNames;
+        },
+
+        /**
+         * Make list of layers in the current document to be used as dropdown options
+         * 
          * @return {Array.<Object>}
          */
-        _getSelectOptions: function () {
-            // get list of layers
+        _getLayerOptions: function () {
+            // Get list of layers
             var appStore = this.getFlux().store("application"),
                 document = appStore.getCurrentDocument(),
                 layers = document.layers.allVisible.reverse(),
                 layerMap = layers.map(function (layer) {
                     // Used to determine the layer face icon
-                    var iconID = "layer-";
-                    if (layer.isArtboard) {
-                        iconID += "artboard";
-                    } else if (layer.kind === layer.layerKinds.BACKGROUND) {
-                        iconID += layer.layerKinds.PIXEL;
-                    } else if (layer.kind === layer.layerKinds.SMARTOBJECT && layer.isLinked) {
-                        iconID += layer.kind + "-linked";
-                    } else {
-                        iconID += layer.kind;
-                    }
-                    
-                    return { id: "layer_" + layer.id.toString(), title: layer.name, type: "item", svgType: iconID };
-                }),
-                layerLabel = { id: "layer_header", title: "Layers", type: "header" },
+                    var iconID = this._getSVGInfo(layer),
+                        ancestry = this._formatLayerAncestry(layer);
+
+                    return {
+                        id: "layer_" + layer.id.toString(),
+                        title: layer.name,
+                        info: ancestry,
+                        displayInfo: ancestry,
+                        svgType: iconID,
+                        type: "item"
+                    };
+                }.bind(this)),
+
+                // Get shortest unique layer ancestry
+                ancestors = collection.pluck(layerMap, "info"),
+                shortenedPaths = pathUtil.getShortestUniquePaths(ancestors.toArray());
+
+            layerMap.map(function (layerItem, index) {
```

Comment:
Use `forEach` for side effects instead of `map`. Use `map` if you're mapping from one array/iterable to another.